R/faceted_qc_plot.R

```diff
@@ -0,0 +1,90 @@
+#' Create a Faceted Scatter Plot for QC
+#'
+#' Generates a customizable, faceted ggplot scatter plot, typically for quality
+#' control (QC). The function supports dynamic labels and automatically
+#' prepares tooltips for interactive use with `ggplotly`.
+#'
+#' @param data A data.frame containing the plotting data.
+#' @param x_var Character. The column name to be used for the x-axis.
+#' @param y_var Character. The column name to be used for the y-axis.
+#' @param colour_var Character. The column name to map to the colour aesthetic.
+#' @param grouping_vars Character vector. Column names used to create vertical
+#'   facets (panels).
+#' @param labels_df A data.frame used by helper functions (`get_label`,
+#'   `generate_tooltip_text`) to look up variable labels.
+#' @param title Character. The main title of the plot.
+#'
+#' @return A `ggplot` object ready to be printed or passed to `ggplotly`.
+#'
+#'
+#' @examples
+#' \dontrun{
+#' # Dummy helper functions required for the example to run
+#' get_label <- function(df, var, type) paste(var, "(units)")
+#' generate_tooltip_text <- function(data, ...) "details"
+#'
+#' # Sample data
+#' qc_data <- data.frame(
+#'   TIME = 1:6,
+#'   RESULT = c(5, 6, 8, 9, 12, 11),
+#'   DOSE = as.factor(c(10, 10, 20, 20, 30, 30)),
+#'   ARM = rep(c("A", "B", "C"), each = 2)
+#'   )
+#' label_data <- data.frame() # Dummy labels object
+#'
+#' # Generate the plot
+#' faceted_qc_plot(
+#'   data = qc_data,
+#'   x_var = "TIME",
+#'   y_var = "RESULT",
+#'   colour_var = "DOSE",
+#'   grouping_vars = "ARM",
+#'   labels_df = label_data,
+#'   title = "Sample QC Plot"
+#'   )
+#' }
+#'
+#' @export
+faceted_qc_plot <- function(data,
+                            x_var,
+                            y_var,
+                            colour_var,
+                            grouping_vars,
+                            labels_df,
```

Comment:
suggestion: `labels_df` could have  default value in case the user does not want to provide labels and wants to rely on variable names.
```suggestion
                            labels_df = data.frame(),
```